presto-main/src/main/java/com/facebook/presto/operator/ContinuousWork.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.presto.operator.ContinuousWorkUtils.WorkState;
+import com.google.common.util.concurrent.ListenableFuture;
+
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+public interface ContinuousWork<T>
+{
+    /**
+     * Call the method to do the work.
+     * The caller can keep calling this method until it returns true.
```

Comment:
This seems counterintuitive that `true` means you can no longer call this method.

Edit: Ok - I take it back. I read a bit more and it seems fine. Just change the javadoc that it is straigtforward that this method is not to just "do the work" but work to prepare result to be returned by `getResult()` later on.